f5lbaasdriver/v2/bigip/driver_v2.py

```diff
@@ -477,7 +477,15 @@ def create(self, context, loadbalancer):
 
             port_data[portbindings.PROFILE] = {}
 
-            vip_masq_mac = device['device_info'].get('masquerade_mac')
+            device_info = device.get('device_info')
+            if not device_info:
+                LOG.error(
+                    "Can not find device info in device %s, when"
+                    " creating loadbalancer %s." % (
+                        device, loadbalancer)
+                )
+
+            vip_masq_mac = device_info.get('masquerade_mac')
```

Comment:
If no device info, your code will continue to run and fails here. There might be two choice:

1. device_info = device.get('device_info', {})

2. remove 'if not device_info:' code block. Since device_info is defined in data model, you can assume it is always there.
3.